core/src/main/java/org/sonar/api/utils/log/SonarLintLogger.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * SonarLint Core - Implementation
+ * Copyright (C) 2009-2017 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.api.utils.log;
+
+import org.sonarsource.sonarlint.core.client.api.common.LogOutput.Level;
+import org.sonarsource.sonarlint.core.log.LogOutputDelegator;
+import org.sonarsource.sonarlint.core.log.MessageFormat;
+
+public class SonarLintLogger extends BaseLogger {
+  private LogOutputDelegator logOutput;
+
+  public SonarLintLogger(LogOutputDelegator logOutput) {
+    this.logOutput = logOutput;
+  }
+
+  @Override
+  void doTrace(String msg) {
+    // no trace logs
```

Comment:
Shouldn't we log to the debug level, just in case?

Reply:
IMO, no. Plugins should know that we don't display trace logs, and if they want things to appear in the logs they should use debug level. They or third party libs that they use might use trace for super detailed things that we definitely would not want to have in sonarlint.